assembly/src/parsers/u32_ops.rs

```diff
@@ -20,9 +20,47 @@ pub fn parse_u32test(span_ops: &mut Vec<Operation>, op: &Token) -> Result<(), As
     Ok(())
 }
 
-/// TODO: implement
-pub fn parse_u32testw(_span_ops: &mut Vec<Operation>, _op: &Token) -> Result<(), AssemblyError> {
-    unimplemented!()
+/// Translates u32testw assembly instruction to VM operation
+/// with a series of DUP U32SPLIT SWAP DROP EQZ on each element in the word.
+pub fn parse_u32testw(span_ops: &mut Vec<Operation>, op: &Token) -> Result<(), AssemblyError> {
+    match op.num_parts() {
+        0 => return Err(AssemblyError::missing_param(op)),
+        1 => {
+            // Test the fourth element
+            span_ops.push(Operation::Dup3);
+            span_ops.push(Operation::U32split);
+            span_ops.push(Operation::Swap);
+            span_ops.push(Operation::Drop);
+            span_ops.push(Operation::Eqz);
+
+            // Test the third element
+            span_ops.push(Operation::Dup3);
+            span_ops.push(Operation::U32split);
+            span_ops.push(Operation::Swap);
+            span_ops.push(Operation::Drop);
+            span_ops.push(Operation::Eqz);
+            span_ops.push(Operation::Or);
```

Comment:
Should this be AND rather than OR?

Reply:
I think it should be OR if we want to have return 1 when there is even one number that's over u32, wdyt?
If we do AND, all of them needs to be over u32 which I thought this is not a "all" but "any" check.

Reply:
If I'm thinking about this correctly, after the `EQZ` operation, the value on the top of the stack will be `1` if the tested value is a u32, and `0` otherwise. Thus, doing `OR` on these values means that if at least one is a u32, the final result will be `1`. But maybe I'm missing something?

Reply:
Ah i see, I think I need to flip the results around, as if I find any number that's not u32, it should keep 1, and or the results together

Reply:
I’m wondering if there (parse_u32testw)
1. miss a `Not` between `Eqz` and `Or`, when ‘testing the second element’ ;
2. miss a `Not` at the end of all 4 tests, for each element if it is u32 the result should be 0(after `Eqz` and `Not`), maybe we should flip the final result, let it be `1` if all elements are u32?

Reply:
Good catch! Thank you for reporting this. I've filed #71 for this.

Reply:
Ah thanks for catching this, fixed this in latest PR